Add TracingConfigFile option to ThanosRuler

Jakob3xD opened the pull request:
## Description

Add the TracingConfigFile option to the Thanos Ruler similar to the Sidecar in in Prometheus spec.
This us useful/needed when the tracing config is a configmap and not a secret.


## Type of change

_What type of changes does your code introduce to the Prometheus operator? Put an `x` in the box that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [ ] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [x] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [ ] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Changelog entry

_Please put a one-line changelog entry below. This will be copied to the changelog file during the release process._

<!-- 
Your release note should be written in clear and straightforward sentences. Most often, users aren't familiar with
the technical details of your PR, so consider what they need to know when you write your release note.

Some brief examples of release notes:
- Add metadataConfig field to the Prometheus CRD for configuring how remote-write sends metadata information.
- Generate correct scraping configuration for Probes with empty or unset module parameter.
-->

```release-note
Add TracingConfigFile option to ThanosRuler
```

simonpasquier reviewed (commented):
Thanks! Could you add a unit test in pkg/thanos/statefulset_test.go next to `TestTracing()`?

Jakob3xD commented:
Hey @simonpasquier,
as requested I added a unit test.
As my knowledge about go is limited, I copied the unit test from the Objectstorage config file and adjusted it for tracing.
I hope this is okay.
